src/compiler/binder.ts

```diff
@@ -278,9 +278,11 @@ namespace ts {
         function declareSymbol(symbolTable: SymbolTable, parent: Symbol, node: Declaration, includes: SymbolFlags, excludes: SymbolFlags): Symbol {
             Debug.assert(!hasDynamicName(node));
 
+            const isJsModuleExport = node.kind === SyntaxKind.BinaryExpression ? getSpecialPropertyAssignmentKind(node) === SpecialPropertyAssignmentKind.ModuleExports : false;

```

Comment:
No need to check `node.kind === SyntaxKind.BinaryExpression`; `getSpecialPropertyAssignmentKind` already does this (and returns `.None`)